File: synth/qmidi.cc
#include "qmidi.h"

typedef unsigned int uint;

QMidiOut::QMidiOut(uint inst, uint vol) {
    // set instruments
    midi.setInst(inst, 0);
    midi.setInst(inst, 1);

    // set volume
    midi.setVol(vol, 0);
    midi.setVol(vol, 1);

    // pitch bend channel 1 up by a quarter tone permanently
    midi.pitchBend(0x2800, 1);
}

QMidiOut::~QMidiOut() {
    // reset pitch bend
    midi.pitchBend(0x2000, 1);
}

File: mini/qmidi.py
import midi

def noteOn(note, vel=127):
    assert 0 <= note <= 255
    midi.noteOn(note // 2, vel, note % 2)

def noteOff(note, vel=127):
    assert 0 <= note <= 255
    midi.noteOff(note // 2, vel, note % 2)

def setVol(vol):
    midi.setVol(vol, 0)
    midi.setVol(vol, 1)

def setInst(inst):
    midi.setInst(inst, 0)
    midi.setInst(inst, 1)